app/scenes/account_notifications/account_notifications.js

```diff
@@ -0,0 +1,518 @@
+        },
+        scrollViewContent: {
+            paddingBottom: 30
+        },
+        wrapper: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg
+        }
+    });
+});
+
+const SAVE_NOTIFY_PROPS = 'save_notify_props';
+const SAVING_NOTIFY_PROPS = 'saving_notify_props';
+
+export default class AccountNofications extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            goBack: PropTypes.func.isRequired,
+            handleUpdateUserNotifyProps: PropTypes.func.isRequired
+        }),
+        config: PropTypes.object.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        myPreferences: PropTypes.object.isRequired,
+        saveRequestStatus: PropTypes.string.isRequired,
+        subscribeToHeaderEvent: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        unsubscribeFromHeaderEvent: PropTypes.func.isRequired
+    };
+
+    static navigationProps = {
+        renderRightComponent: (props, emitter) => {
+            return <SaveNotifcationsButton emitter={emitter}/>;
+        }
+    }
+
+    constructor(props) {
+        super(props);
+
+        const {currentUser} = props;
+        const notifyProps = currentUser.notify_props || {};
+        this.setStateFromNotifyProps(notifyProps);
+    }
+
+    componentWillMount() {
+        this.props.subscribeToHeaderEvent(SAVE_NOTIFY_PROPS, this.saveUserNotifyProps);
+    }
+
+    componentWillUnmount() {
+        this.props.unsubscribeFromHeaderEvent(SAVE_NOTIFY_PROPS);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.currentUser !== this.props.currentUser) {
+            const {notify_props: notifyProps} = nextProps.currentUser;
+            this.setStateFromNotifyProps(notifyProps);
+        }
+
+        if (nextProps.saveRequestStatus === RequestStatus.SUCCESS && this.props.saveRequestStatus === RequestStatus.STARTED) {
+            this.props.actions.goBack();
+        } else if (nextProps.saveRequestStatus === RequestStatus.FAILURE && this.props.saveRequestStatus === RequestStatus.STARTED) {
+            EventEmitter.emit(SAVING_NOTIFY_PROPS, false);
+            this.setStateFromNotifyProps(this.currentUser.notify_props);
+        }
+    }
+
+    setStateFromNotifyProps = (notifyProps) => {
+        const mentionKeys = (notifyProps.mention_keys || '').split(',');
+        const usernameMentionIndex = mentionKeys.indexOf(this.props.currentUser.username);
+        if (usernameMentionIndex > -1) {
+            mentionKeys.splice(usernameMentionIndex, 1);
+        }
+
+        let email = notifyProps.email;
+        if (this.props.config.EnableEmailBatching === 'true') {
+            const emailPreferences = getPreferencesByCategory(this.props.myPreferences, Preferences.CATEGORY_NOTIFICATIONS);
+            const interval = emailPreferences.get(Preferences.EMAIL_INTERVAL).value;
+            email = interval;
```

Comment:
We should use two different fields for the email interval and whether or not its enabled. This seems prone to errors.